### PR TITLE
Fix for supply reorder widget

### PR DESCRIPTION
**Pull request by radavis**
## Summary

The supply reorder page requires the ctaWidget to display a `h4` heading, and not a `h3` heading.

- Adds optional `headingLevel` prop to `src/applications/static-pages/cta-widget/ctaWidgets.js`
- Updates unit specs

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#79823
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#29499

## Testing done

Steps to verify in local development:

1. `yarn build` from `vets-website`
1. `yarn watch` from `content-build`
1. `foreman start -m all=1,clamd=0,freshclam=0` from `vets-api` branch `master`
1. Add HTML to `content-build/build/localhost/health-care/order-hearing-aid-batteries-and-accessories/index.html`: `<div data-template="paragraphs/react_widget" class="cta-widget" data-widget-type="cta" data-app-id="ha-cpap-supplies"></div>` Note that this step is done through the Drupal content editor for other environments.
1. Visit: `http://localhost:3002/health-care/order-hearing-aid-batteries-and-accessories/#how-do-i-order-hearing-aid-bat`, observe that no content has changed.
1. Visit: `http://localhost:3000/flipper/features/ha_cpap_supplies_cta` and click "Enable for everyone"
1. Visit: `http://localhost:3002/health-care/order-hearing-aid-batteries-and-accessories/#how-do-i-order-hearing-aid-bat`, observe that content _has_ changed.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |    Screenshot 2024-05-13 at 14-56-00 Order hearing aid batteries and accessories from VA Veterans Affairs    |    Screenshot 2024-05-13 at 14-55-20 Order hearing aid batteries and accessories from VA Veterans Affairs   |

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added

### CODE REVIEW

File: src/applications/static-pages/cta-widget/components/messages/Verify.jsx [hunk 1]
```diff
@@ -27,8 +28,9 @@ const Verify = ({ serviceDescription, primaryButtonHandler }) => {
 };
 
 Verify.propTypes = {
-  serviceDescription: PropTypes.string.isRequired,
   primaryButtonHandler: PropTypes.func.isRequired,
+  serviceDescription: PropTypes.string.isRequired,
+  headerLevel: PropTypes.number,
```
Comment (carlosfelix2) [inline src/applications/static-pages/cta-widget/components/messages/Verify.jsx:33]: There is a mismatch here. Header level is a number here, but a string [below](https://github.com/department-of-veterans-affairs/vets-website/pull/29702/files#diff-064b584551b9f79877de15d42d64e617ebc3cfbd2e7414957768c9b04f2843ffL47). Can that cause any issues?

**Review by carlosfelix2 — DISMISSED** [review]
Missing complete PR description including screenshots.

File: src/applications/static-pages/cta-widget/components/messages/Verify.jsx (hunk 1 above)
Comment (radavis) [reply src/applications/static-pages/cta-widget/components/messages/Verify.jsx:33]: Yeah, proptypes will complain. Will change.

Comment (radavis) [reply src/applications/static-pages/cta-widget/components/messages/Verify.jsx]: Latest changes have screenshots added and the prop-types:

```
Verify.propTypes = { headerLevel: PropTypes.string };  // no change
SignIn.propTypes = { headerLevel: PropTypes.oneOfType([PropTypes.number, PropTypes.string]);  // string added
CallToActionWidget.propTypes = { headerLevel: PropTypes.string }; // no change
```

**Review by carlosfelix2 — APPROVED** [review]

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**